### DISCUSSION TITLE
IOS Safari requestIdleCallback problem

### DISCUSSION THREAD

**Discussion by weboose** [breezefront/community#215] [1. Help]
I've discovered that window.requestIdleCallback doesn't work well on some versions of iOS Safari.
The following code will be executed, but it doesn't actually do anything, the callback is not called.
```
if (window.requestIdleCallback) {
    window.requestIdleCallback(callback);
}
```

**Comment by vovayatsyuk** [comment]
Which safari version? What do you mean by "code will be executed"? What happens if you write this:

```js
if (window.requestIdleCallback) {
    window.requestIdleCallback(callback);
} else {
    window.setTimeout(callback, 0);
}
```

**Reply by weboose** [reply]
IOS 17
Safari: unknown
if(window.requestIdleCallback) is established, but window.requestIdleCallback does not call callback

**Reply by vovayatsyuk** [reply]
Safari doesn't support this feature yet: https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback#browser_compatibility. 

`window.requestIdleCallback` should return `undefined`.

I think that you have some polyfill script included on the page that doesn't work well. Try to search for `window.requestIdleCallback =` across your site.
